jwt_vc_json/jwt_vc_json/v1_0/cred_processor.py

```diff
@@ -0,0 +1,63 @@
+"""Issue a jwt_vc_json credential."""
+
+import datetime
+import logging
+import uuid
+
+from aiohttp import web
+from aries_cloudagent.admin.request_context import AdminRequestContext
+from aries_cloudagent.wallet.jwt import jwt_sign
+
+from oid4vci.models.exchange import OID4VCIExchangeRecord
+from oid4vci.models.supported_cred import SupportedCredential
+from oid4vci.public_routes import types_are_subset, PopResult, ICredProcessor
+
+LOGGER = logging.getLogger(__name__)
+
+
+class CredProcessor(ICredProcessor):
+    """Credential processor class for jwt_vc_json format."""
+
+    async def issue_cred(
+        self,
+        body: any,
+        supported: SupportedCredential,
+        ex_record: OID4VCIExchangeRecord,
+        pop: PopResult,
+        context: AdminRequestContext,
+    ):
+        """Return signed credential in JWT format."""
+        if not types_are_subset(body.get("types"), supported.format_data.get("types")):
+            raise web.HTTPBadRequest(reason="Requested types does not match offer.")
```

Comment:
I would recommend raising an exception not specific to the context of HTTP Handlers.

The module defining the `ICredProcessor` could, for instance, define a `CredIssueError` or something that each implementing class could raise for these kinds of errors.

Then, when the cred processors are called from an HTTP handler, these exceptions should be caught and turned into HTTPBadRequest, etc.

Reply:
Good point, thanks for that. I'll make change.